### ISSUE TITLE
MzTabExporter output, wrong number of peptide_abundance_study_variable columns in PEP section 

### ISSUE THREAD

**Issue by lars20070** [OpenMS/OpenMS#1456]
When exporting peptide feature pairs to mzTab, the PEP sections contains columns such as peptide_abundance_study_variable[1] and peptide_abundance_study_variable[2].

But there appears also a third column peptide_abundance_study_variable[3] with all null values which is not necessary.

**Comment by timosachsenberg** [comment]
Could you please add a gist of the consensusXML so I can take a look at it.

**Comment by lars20070** [comment]
Here the original consensusXML. https://gist.github.com/lars20070/0b14d37df961fbdcd3d6

**Comment by lars20070** [comment]
Please replace spaces by underscores in opt_psm_Posterior Error Probability_score.

**Comment by lars20070** [comment]
spectra_ref, database, database_version, search_engine still missing.

**Comment by lars20070** [comment]
And the original search engine score (MSGF+) would be useful #1453.

**Comment by timosachsenberg** [comment]
the number of columns are indeed wrong.
What about the other issues? These are not related to the consensusXML you provided aren't they?

**Comment by lars20070** [comment]
I updated the gist above. The cXML now also contains the ID information.

**Comment by timosachsenberg** [comment]
search_engine is set to OpenMS/ConsensusID_best so I could export this informatin.
The other information is not set in the consensusXML so MzTabExporter can't export it.

```
<IdentificationRun id="PI_0" date="2015-05-29T15:33:26" search_engine="OpenMS/ConsensusID_best" search_engine_version="2.0.0">
		<SearchParameters db="" db_version="" taxonomy="" mass_type="monoisotopic" charges="" enzyme="unknown_enzyme" missed_cleavages="0" precursor_peak_tolerance="0" peak_mass_tolerance="0" >
		</SearchParameters>
		<ProteinIdentification score_type="q-value" higher_score_better="false" significance_threshold="0">
			<ProteinHit id="PH_0" accession="sp|A2ADY9|DDI2_MOUSE" score="0.970492362976074" 				<userParam type="string" name="target_decoy" value="target"/>
				<userParam type="string" name="Description" value="Protein DDI1 homolog 2 OS=Mus musculus GN=Ddi2 PE=1 SV=1"/>
				<userParam type="float" name="_score" value="0"/>
			</ProteinHit>
```

**Comment by timosachsenberg** [comment]
slightly of topic but does anybody know what name="_score" is?

**Comment by lars20070** [comment]
An optional column opt_map with the map/fraction information would be helpful.

Here the workflow as reference. https://gist.github.com/lars20070/3efb70bb333ac6a87665
Either PeptideIndexer does not annotate the database information or it's not passed on in one of the downstream tools.

**Comment by timosachsenberg** [comment]
what do you mean with opt_map and how would that play with linked label-free files?
Where should this information come from?

**Comment by timosachsenberg** [comment]
"Please replace spaces by underscores in opt_psm_Posterior Error Probability_score." 
this is how it is annotated in the consensusXML so I won't do any editing in MzTabExporter. One might think about changing it in the original tool (maybe with CV etc.).

**Comment by timosachsenberg** [comment]
"And the original search engine score (MSGF+) would be useful #1453."
I think it is not contained in the consensusXML anymore. So let's determine the actual issue (e.g. tool X discards MetaInformation)

**Comment by timosachsenberg** [comment]
"spectra_ref" is not stored in peptide hit so the spectrum index cannot be exported.
Do we have the information from which map the id originates? Because that information is also important to link to the actual file.

**Comment by lars20070** [comment]
A further opt_fraction column would be good. For the example above, either WL383, WL384, ... WL388. It's almost always multiple-fraction-experiments. RT and mz are only useful if one knows which fraction this refers to.

I know most of the missing information is not even in the cXML. We can discuss this in separate issues.

**Comment by lars20070** [comment]
"Do we have the information from which map the id originates?"

Yes, optional meta value "map index" in peptide id. From annotate_ids_with_subelements in IDMapper.

**Comment by timosachsenberg** [comment]
partially adressed by https://github.com/OpenMS/OpenMS/pull/1459

**Comment by lars20070** [comment]
Original issue resolved by #1459. Remaining issues documented in #1453.
